app/views/companies/show.html.haml

```diff
@@ -27,8 +27,8 @@
 
         = field_or_none t('.website'), link_to(@company.website, sanitize(full_uri(@company)), target: '_blank'), tag_options: {class: 'website'}
 
-        - if current_user.is_in_company_numbered?(@company.company_number) ||  |
-             current_user.admin?                                               |
+        - if (current_user.is_in_company_numbered?(@company.company_number) && |
+              beta_user?(current_user)) || current_user.admin?                 |
 
```

Comment:
This would be slightly faster if you put the `current_user.admin?` first, now that you've added the `&& beta_user?(current_user)` condition.

(But really should be using a Pundit method to see if they're authorized to see/use 'beta testers only' stuff.)